ipatests/test_integration/test_user_permissions.py

```diff
@@ -93,3 +97,55 @@ def test_stageuser_show_as_alternate_admin(self):
         # the field Kerberos Keys available must contain True
         result = self.master.run_command(['ipa', 'stageuser-show', stageuser])
         assert 'Kerberos keys available: True' in result.stdout_text
+
+
+class TestInstallClientNoAdmin(IntegrationTest):
+    num_clients = 1
+
+    def test_installclient_as_user_admin(self):
+        """ipa-client-install should not use hardcoded admin for principal
+
+        In ipaclient-install.log it should use the username that was entered
+        earlier in the install process at the prompt.
+        Related to : https://pagure.io/freeipa/issue/5406
+        """
+        client = self.clients[0]
+        tasks.install_master(self.master)
+        tasks.kinit_admin(self.master)
+        username = 'testuser1'
+        password = 'userSecretPassword123'
+        password_confirmation = "%s\n%s\n" % (password,
+                                              password)
+
+        self.master.run_command(['ipa', 'user-add', username,
+                                 '--first', username,
+                                 '--last', username,
+                                 '--password'],
+                                stdin_text=password_confirmation)
+
+        role_add = ['ipa', 'role-add', 'useradmin']
+        self.master.run_command(role_add)
+        self.master.run_command(['ipa', 'privilege-add', 'Add Hosts'])
+        self.master.run_command(['ipa', 'privilege-add-permission',
+                                 '--permissions', 'System: Add Hosts',
+                                 'Add Hosts'])
+
+        self.master.run_command(['ipa', 'role-add-privilege', 'useradmin',
+                                 '--privileges', 'Host Enrollment'])
+
+        self.master.run_command(['ipa', 'role-add-privilege', 'useradmin',
+                                 '--privileges', 'Add Hosts'])
+
+        role_member_add = ['ipa', 'role-add-member', 'useradmin',
+                           '--users=' + username]
```

Comment:
@amore17  please use .format() :) I forgot about this one

Reply:
@Tiboris I have added changes as per your review :)